mne/surface.py

```diff
@@ -1840,28 +1824,40 @@ def warp_montage_volume(montage, base_image, reg_affine, sdr_morph,
     _check_subject_dir(subject_to, subjects_dir)
 
     # load image and make sure it's in surface RAS
-    image, fs_t1 = _get_surface_RAS_volumes(
-        base_image, subject_from, subjects_dir)
+    if not isinstance(base_image, nib.spatialimages.SpatialImage):
+        base_image = nib.load(base_image)
+    fs_from_img = nib.load(
+        op.join(subjects_dir, subject_from, 'mri', 'brain.mgz'))
+    if not np.allclose(base_image.affine, fs_from_img.affine, atol=1e-6):
+        raise RuntimeError('The `base_image` is not aligned to Freesurfer '
+                           'surface RAS space. This space is required as '
+                           'it is the space where the anatomical '
+                           'segmentation and reconstructed surfaces are')
 
     # get montage channel coordinates
     ch_dict = montage.get_positions()
     if ch_dict['coord_frame'] != 'mri':
+        bad_coord_frames = np.unique([d['coord_frame'] for d in montage.dig])
+        bad_coord_frames = ', '.join([
+            _frame_to_str[cf] if cf in _frame_to_str else str(cf)
+            for cf in bad_coord_frames])
         raise RuntimeError('Coordinate frame not supported, expected '
-                           '"mri", got ' + str(ch_dict['coord_frame']))
-    ch_coords = np.array(list(ch_dict['ch_pos'].values()))
+                           f'"mri", got {bad_coord_frames}')
+    ch_names = list(ch_dict['ch_pos'].keys())
+    ch_coords = np.array([ch_dict['ch_pos'][name] for name in ch_names])
 
     # convert to freesurfer voxel space
     ch_coords = apply_trans(
-        np.linalg.inv(fs_t1.header.get_vox2ras_tkr()), ch_coords * 1000)
+        np.linalg.inv(fs_from_img.header.get_vox2ras_tkr()), ch_coords * 1000)
 
     # take channel coordinates and use the image to transform them
     # into a volume where all the voxels over a threshold nearby
     # are labeled with an index
-    image_data = _get_img_fdata(image)
+    image_data = np.array(base_image.dataobj)
```

Comment:
`np.array` is okay here as you want a copy anyway. But keep in mind that the values can now in general be non-float (and have over/underflow problems like `np.uint8`), unlike `_get_img_fdata`. So I actually think `_get_img_fdata` might be a safer choice here. The existing code will probably break quite badly if the dtype is `np.uint8` and there are more than 255 electrodes...